app/models/credential.rb

```diff
@@ -7,33 +7,56 @@ class Credential < ApplicationRecord
 
   before_create :set_expires_at
 
-  def logged_in?
-    jwt_payload["login"].present?
-  rescue JWT::ExpiredSignature
+  def admin?
+    return true if Rails.env.development?
+
+    if @admin.nil?
+      client.authenticated_admin?
+    else
+      @admin
+    end
+  rescue Panoptes::Client::NotLoggedIn, Panoptes::Client::AuthenticationExpired
     false
   end
 
-  def login
-    jwt_payload.fetch("login")
+  def authenticate!
+    client.authenticated?
   end
 
-  def user_id
-    jwt_payload["id"]
+  def display_name
+    client.authenticated_user_display_name
   end
 
-  def admin?
+  def expired?
+    expires_at < Time.zone.now
+  end
+
+  def logged_in?
     return true if Rails.env.development?
-    jwt_payload.fetch("admin", false)
-  rescue JWT::ExpiredSignature
+
+    if @logged_in.nil?
+      client.authenticated?
+    else
+      @logged_in
+    end
+  rescue Panoptes::Client::NotLoggedIn, Panoptes::Client::AuthenticationExpired
     false
   end
 
-  def ok?
-    jwt_payload.present? && !expired?
+  def login
+    if @login.nil?
+      client.authenticated_user_login
+    else
+      @login
+    end
   end
 
-  def expired?
-    expires_at < Time.zone.now
+  def user_id
+    if @user_id.nil?
```

Comment:
I think these are being set in your test suite through the transient evaluator but not in the actual code base...thoughts?

Reply:
the instance variable exists to let me create admin credentials using the factory, otherwise it is unused.